cmd/argocd/commands/admin/backup_test.go

```diff
@@ -12,6 +20,32 @@ import (
 	"github.com/argoproj/argo-cd/v3/common"
 )
 
+func createTestWriter(t *testing.T, out string) (io.Writer, func()) {
+	t.Helper()
+
+	var writer io.Writer
+	var cleanup func()
+
+	if out == "-" {
```

Comment:
Instead of passing in a string "out", should the user of these functions pass in an `io.Writer`? This function also wouldn't need to have any cleanup logic then.

Reply:
I removed the logic for creating the test writer and check for in-memory buffer instead. This reduces the code and overall complexity as well. Please let me know what do you think about it.